### PR TITLE
feat: set-up docsite

### PR DESCRIPTION
**Pull request by nayib-jose-gloria**
## Reason for Change

- #11

## Changes

- renamed constants.py to _constants.py to not expose constants in API docs
- use pdoc to generate documentation based on docstring
- add gha to deploy docsite changes on merge to main

## Testing steps
- update GHA to test that docsite won't break as a 'push test', but only deploy new site on merge to main

## Notes for Reviewer
- tidying up docstrings, adding more README infos and markdowns
- versioning support > not including at this time.
- pdoc docs: https://pdoc.dev/docs/pdoc.html#add-documentation

### CODE REVIEW

File: api/python/Makefile [hunk 1]
```diff
@@ -4,6 +4,10 @@ install: package-data
 install-dev: package-data
 	pip install -e .[test]
 
+install-docs:
+	pip install .
+	pip install pdoc
```
Comment (Bento007) [inline api/python/Makefile:9]: add this to pyproject.yaml
```python
[project.optional-dependencies]
test = ["pytest"]
docs = ["pydoc"]
```
Then change this make recipe to
```make
install-docs:
	pip install .[docs]
```

**Comment by Bento007** [comment]
I made a sphinx version of the docs, and it was a lot more work and still doesn't look as good as the pdoc version. If we host on readthedocs we get the versioning for free, but that requires a subscription. Sphinx doesn't support it out of the box and will require additional configuration to work. For now I think this is good enough. We should host the latest at least and can add instructions that the users can follow to generate docs for older versions.

**Review by nayib-jose-gloria — COMMENTED** [review]
not sure where the docsite homepage description is being set, but there is a typo:

"CellxGene Ontology Guide is a filtered and _currated_ collection of ontological metadata from different public sources"

**Review by nayib-jose-gloria — COMMENTED** [review]
Data submodule page is empty on the docsite as well, we should probably find a way to not include it since its an internal directory anyway

**Review by nayib-jose-gloria — COMMENTED** [review]
https://chanzuckerberg.github.io/cellxgene-ontology-guide/cellxgene_ontology_guide/ontology_parser.html#OntologyParser.cxg_schema

weird non-definition for this attribute here

**Review by nayib-jose-gloria — COMMENTED** [review]
non-definitions of attributes here as well https://chanzuckerberg.github.io/cellxgene-ontology-guide/cellxgene_ontology_guide/supported_versions.html#CXGSchema.supported_ontologies

**Review by nayib-jose-gloria — COMMENTED** [review]
looks good overall, just a few things to fix

In a separate ticket, we should document (in the repo, not docsite) the release process + changelog best practices for contributors

**Review by nayib-jose-gloria — COMMENTED** [review]
LGTM!

**Pull request merged**